### PR TITLE
DB, ticket router update

### PR DESCRIPTION
**Pull request by arasmu2**
## Brief Description

Added instructions to connect PGAdmin to the docker DB

Updated the DB to mark only instances from the current season as active

Fixed the SQL query that returns tickets to active instances
-for rows returned by this query, fixed the date parsing to account for the new DB

## References Issue

Partial resolution of Issue #115

## Type of change

- [x] Bug fix
- [ ] Feature
- [ ] Major change (potentially breaking, list anything that is broken below)
- [ ] Documentation

### Breaking Changes

None

## Testing done

- [x] Docker was used.
- [ ] Docker was not used and I have listed my configuration below.

**System Specifications**:
* Operating System:
* Node Version:
* NPM Version:

## Checklist:

- [x] My code follows the styling guidelines.
- [ ] I have added unit tests and verified that they pass.
- [x] I have used the linter and fixed any linting issues.
- [x] I have commented the code.
- [ ] I have adjusted the documentation to match the changed code.
- [x] Prior to submitting the PR, I made sure the latest changes were merged with my branch.
- [ ] Existing tests pass locally with changes.

### CODE REVIEW

File: task_sql/notes.txt [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+  Open PGAdmin and select 'Add New Server'
+    Under the 'General' tab, give the server a name
+    Under the 'Connection' tab:
+      Set hostname/address to 'localhost'
+      Verify Port is set to 5432
+      Set username/password to the same values in your .env file (postgres default for both)
+  Click save and you will be connected
+
+
+To connect to the DB from docker command line:
```
Comment (ben16williams) [inline task_sql/notes.txt:18]: Docker command line is a little ambiguous, I would suggest elaborating here more e.g. "From Docker Desktop, go to the server container and select 'container actions' -> 'open in terminal'" or something like that

**Comment by arasmu2** [comment]
@ben16williams pinging you for a review/approval

File: server/src/api/events/event.service.ts [hunk 2]
```diff
@@ -5,38 +5,37 @@ import {pool, response, buildResponse} from '../db';
...
+          eventid id, 
           seasonid, 
           eventname title, 
           eventdescription description, 
           active, 
           imageurl image_url, 
           count(eventinstances.eventinstanceid) as numShows
-          FROM events natural join eventinstances
-          GROUP BY events.eventid,
+        FROM events natural join eventinstances
```
Comment (ben16williams) [inline server/src/api/events/event.service.ts:16]: I think we should avoid using NATURAL JOIN, it makes reading the query more difficult unless you know all the fields of the table. I think an equivalent command would be `FROM eventinstances JOIN events ON eventid` if the schema I'm looking at is current. The commands should also be capitalized for consistency

File: server/src/api/events/event.service.ts [hunk 3]
```diff
@@ -201,37 +216,34 @@ export const checkIn = async (params: any): Promise<response> => {
...
+          eventinstanceid,
+          eventid,
+          eventname,
+          eventdescription,
+          imageurl,
+          eventdate, 
+          eventtime,
+          totalseats,
+          availableseats
+        FROM eventinstances NATURAL JOIN events
```
Comment (ben16williams) [inline server/src/api/events/event.service.ts:229]: Same here

**Review by ben16williams — CHANGES_REQUESTED** [review]

**Comment by ben16williams** [comment]
This is totally optional but you could also modify setup.py so that the default database name is `wondertix`

File: server/src/api/events/event.service.ts (hunk 2 above)
Comment (arasmu2) [reply server/src/api/events/event.service.ts:16]: I agree, it's probably a better idea to be explicit about the joins.

**Comment by arasmu2** [comment]
Can I get these changes merged for the moment? I'm updating the field names in all tables to explicitly denote a field as a foreign keys (eg: `eventid_fk`) which will require explicit joins instead of natural joins. I will also set the default DB name as `wondertix`

**Comment by ben16williams** [comment]
> Can I get these changes merged for the moment? I'm updating the field names in all tables to explicitly denote a field as a foreign keys (eg: `eventid_fk`) which will require explicit joins instead of natural joins. I will also set the default DB name as `wondertix`

Sure, just make sure to put it in a ticket

**Review by ben16williams — APPROVED** [review]

**Pull request merged**